src/App.ts

```diff
@@ -568,18 +574,22 @@ function singleTeamAuthorization(
   authorization: Partial<AuthorizeResult> & { botToken: Required<AuthorizeResult>['botToken'] },
 ): Authorize {
   // TODO: warn when something needed isn't found
-  const botUserId: Promise<string> = authorization.botUserId !== undefined ?
-    Promise.resolve(authorization.botUserId) :
+  const auth: Promise<AuthConstraints> = authorization.botUserId !== undefined && authorization.botId !== undefined ?
```

Comment:
Do we see any potential use of `AuthConstraints` outside of this method's implementation details? If not, then I'd prefer if we just declared an inline interface here instead of the named interface at the top of the file
```suggestion
  const auth: Promise<{ botUserId: string, botId: string }> = authorization.botUserId !== undefined && authorization.botId !== undefined ?
```